R/derive_param_framingham.R

```diff
@@ -0,0 +1,370 @@
+#'   HDL Chol Factor = -0.70833;
+#'   SysBPFactor=2.76157 if not on hypertension medications, 2.82263 if on
+#'     hypertension medications, Avg Risk = 26.1931 and Risk Period;
+#'   Smoker=.52873 if smoker, 0 otherwise;
+#'   Diabetes Present=.69154 is present, 0 otherwise;
+#'   Avg Risk = 26.1931;
+#'   RiskPeriodFactor = 0.95012;
+#'
+#' For men:
+#'   Age Factor = 3.06117;
+#'   Total Chol Factor = 1.12370;
+#'   HDL Chol Factor = -0.93263;
+#'   SysBPFactor=1.93303 if not on hypertension medications, 2.99881 if on
+#'     hypertensino medications,;
+#'   Smoker=.65451 if smoker, 0 otherwise;
+#'   Diabetes Present=.57367 is present, 0 otherwise,;
+#'   Avg Risk = 23.9802;
+#'   Risk Period Factor = 0.88936;
+#'
+#' Equation:
+#' \deqn{RiskFactors = (log(Age) * AgeFactor)
+#' + (log(TotalChol) * TotalCholFactor)
+#' + (log(CholHDL) * CholHDLFactor)
+#' + (log(SysBP) * SysBPFactor) + Smoker
+#' + Diabetes Present - AvgRisk}
+#'
+#' \deqn{Risk = 100 * (1 - RiskPeriodFactor ^ exp(RiskFactors))}
+#'
+#' @return A numeric vector of Framingham values
+#'
+#' @keywords computation
+#'
+#' @export
+#'
+#' @examples
+#' compute_framingham(sysbp = 133, chol = 216.16, cholhdl = 54.91, age=53,
+#'                    sex = "M", smokefl = "N", diabetfl = "N", trthypfl = "N")
+#'
+#' compute_framingham(sysbp = 161, chol = 186.39, cholhdl = 64.19, age=52,
+#'                    sex = "F", smokefl = "Y", diabetfl = "N", trthypfl = "Y")
+#'
+compute_framingham <- function(sysbp, chol, cholhdl, age, sex, smokefl,
+                               diabetfl, trthypfl) {
+
+  assert_numeric_vector(sysbp)
+  assert_numeric_vector(chol)
+  assert_numeric_vector(cholhdl)
+  assert_numeric_vector(age)
+  assert_character_vector(sex, values = c("M", "F"))
+  assert_character_vector(smokefl, values = c("Y", "N"))
+  assert_character_vector(diabetfl, values = c("Y", "N"))
+  assert_character_vector(trthypfl, values = c("Y", "N"))
+
+  aval <- case_when(
+    sex == "F" ~
+      1 - (0.95012 ^ exp((2.32888 * log(age))
+                         + (1.20904 * log(chol))
+                         - (0.70833 * log(cholhdl))
+                         + (2.76157 * log(if_else(trthypfl == "N", sysbp, 1)))
+                         + (2.82263 * log(if_else(trthypfl == "Y", sysbp, 1)))
+                         + (0.52873 * (if_else(smokefl == "Y", 1, 0)))
+                         + (0.69154 * (if_else(diabetfl == "Y", 1, 0)))
+                         - 26.1931)),
+    sex == "M" ~
+      1 - (0.88936 ^ exp((3.06117 * log(age))
+                         + (1.12370 * log(chol))
+                         - (0.93263 * log(cholhdl))
+                         + (1.93303 * log(if_else(trthypfl == "N", sysbp, 1)))
+                         + (1.99881 * log(if_else(trthypfl == "Y", sysbp, 1)))
+                         + (0.65451 * (if_else(smokefl == "Y", 1, 0)))
+                         + (0.57367 * (if_else(diabetfl == "Y", 1, 0)))
+                         - 23.9802)))
```

Comment:
Do we need all these brackets?

Reply:
Technically not but I think it makes the code cleaner